### ISSUE TITLE
imageStore() fails to validate

### ISSUE THREAD

**Issue by dzil123** [gfx-rs/naga#918]
The following shader fails to validate on wgpu v0.8.1 + naga v0.4.1 with `ShaderFlags::VALIDATION`: 
```glsl
#version 450

layout(local_size_x = 8, local_size_y = 8) in;

layout(set = 0, binding = 0, rgba8) writeonly uniform image2D output_tex;

void main() {
    imageStore(output_tex, ivec2(gl_GlobalInvocationID.xy), vec4(0));
}
```
This is the error:
```
wgpu error: Validation Error

Caused by:
    In Device::create_shader_module
      note: label = `shader.comp`
    Function [1] '' is invalid
    Expression [14] is invalid
    Used by a statement before it was introduced into the scope by any of the dominating blocks
```

Here is the compiled shader: [shader.comp.spirv.zip](https://github.com/gfx-rs/naga/files/6558578/shader.comp.spirv.zip)

**Comment by kvark** [comment]
Fixed by #876, which just needs to be back-ported into a patch.

**Comment by dzil123** [comment]
Fixed by v0.4.2. Thank you!
